### PR TITLE
test: update cli tests to use JWT-based authentication

### PR DESCRIPTION
**Pull request by menuka94**
## Purpose
$subject

### CODE REVIEW

File: integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/Constants.java [hunk 1]
```diff
@@ -20,13 +20,16 @@
...
+    public static final String DEFAULT_USERNAME = "admin";
+    public static final String DEFAULT_PASSWORD = "admin";
     static final String API = "api";
     static final String SHOW = "show";
     static final String COMPOSITAPP = "compositeapp";
     static final String ENDPOINT = "endpoint";
     static final String PROXYSERVICE = "proxyservice";
     static final String SEQUENCE = "sequence";
-    static final String REMOTE = "remote";
+    public static final String REMOTE = "remote";
```
Comment (sdkottegoda) [inline integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/Constants.java]: Any reason to have only some constants as public? Or is being package private enough?

File: integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliProxyserviceTestCase.java [hunk 2]
```diff
@@ -39,7 +40,7 @@ public void miShowProxyAllTest() throws IOException {
         String artifactName_proxy_1[] = TestUtils.getArtifactList(outputForCLICommand).get(0).split(" ", 2);
         String artifactName_proxy_2[] = TestUtils.getArtifactList(outputForCLICommand).get(1).split(" ", 2);
 
-        Assert.assertEquals(artifactName_proxy_1[0], CLI_ADDRESS_PROXY);
+        Assert.assertEquals(artifactName_proxy_1[0], CLI_MEDIATOR_PROXY);
```
Comment (menuka94) [inline integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliProxyserviceTestCase.java]: This change was made to fix an already existing test failure and is unrelated to the purpose of this PR.

**Comment by Nirothipan** [comment]
The conf dir can be removed as well, no need for .gitkeep

File: integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliEnvironmentSetup.java [hunk 3]
```diff
@@ -48,4 +49,9 @@ public void setupEnv() throws IOException {
             }
         }
     }
+
+    @BeforeSuite
+    public void login() throws IOException {
```
Comment (Nirothipan) [inline integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliEnvironmentSetup.java]: doesn't this method needs to depends on setupEnv

File: integration/cli-tests/src/test/java/util/TestUtils.java [hunk 4]
```diff
@@ -23,13 +23,26 @@
...
     }
 
+    public static void login() throws IOException {
+        LOG.info("Logging in to use Micro Integrator Management API");
+        // mi remote login admin admin
+        String[] arguments = new String[]{getMIBuildPath(), Constants.REMOTE, Constants.LOGIN,
+                Constants.DEFAULT_USERNAME, Constants.DEFAULT_PASSWORD};
+        runCommandWithArgs(arguments);
+    }
+
```
Comment (Nirothipan) [inline integration/cli-tests/src/test/java/util/TestUtils.java]: Shall we try log out as well after class and try to invoke something after logout and check whether it fails

File: integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliEnvironmentSetup.java (hunk 3 above)
Comment (menuka94) [reply integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/CliEnvironmentSetup.java]: Updated.

File: integration/cli-tests/src/test/java/util/TestUtils.java (hunk 4 above)
Comment (menuka94) [reply integration/cli-tests/src/test/java/util/TestUtils.java]: Updated.

File: integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/LoginLogoutTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+public class LoginLogoutTest extends AbstractCliTest {
+
+    @Test
+    public void loginTest() throws IOException {
+        List<String> loginOutput = super.login();
+        Assert.assertTrue(loginOutput.get(0).contains("Login successful for remote"));
+    }
+
+    @Test(dependsOnMethods = {"loginTest"})
```
Comment (sdkottegoda) [inline integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/LoginLogoutTest.java]: Better if we don't have to depend on methods, rather make methods self contained.

Comment (menuka94) [reply integration/cli-tests/src/test/java/org/wso2/micro/integrator/cli/LoginLogoutTest.java]: Updated.

**Pull request merged**